os/drivers/lcd/mipi_lcd.c

```diff
@@ -254,6 +254,34 @@ static int send_init_cmd(struct mipi_lcd_dev_s *priv, lcm_setting_table_t *table
 	}
 }
 
+static int check_lcd_vendor_send_init_cmd(struct mipi_lcd_dev_s *priv)
+{
+#ifdef CONFIG_LCD_ST7785
+	/* Check Vendor */
+	lcm_setting_table_t read_display_cmd = {0x04, 0, {0x00}};
+	uint8_t rxbuf[3] = {0xFF,0xFF,0xFF};
+	int status;
+	status = set_return_packet_len(priv, sizeof(rxbuf)/sizeof(rxbuf[0]));
+	if (status != OK) {
+		return status;
+	}
+	status = read_response(priv, read_display_cmd, rxbuf, sizeof(rxbuf)/sizeof(rxbuf[0]));
+	if (status != OK) {
+		return status;
+	}
+	for (int i = 0; i < sizeof(rxbuf)/sizeof(rxbuf[0]); i++) {		/* LCD info */
+		lcdvdbg("%2x\n", rxbuf[i]);
+	}
+	if (rxbuf[2] == 0x01) { /* checking 0xDC used by AVD */
+		return send_init_cmd(priv, lcd_init_cmd_g_avd);
+	}
+	if (rxbuf[0] == 0x02) { /* checking 0xDA used by Holitech */
+		return send_init_cmd(priv, lcd_init_cmd_g_holitech);
+	}
+#endif
+	return send_init_cmd(priv, lcd_init_cmd_g);
```

Comment:
We should manage the files for common and for some specific LCD separately. 
This kind of coding is difficult to maintain. Could you do refactoring?